common/src/main/java/io/pravega/common/tracing/RequestTracker.java

```diff
@@ -170,6 +174,9 @@ public void trackRequest(String requestDescriptor, long requestId) {
             }
 
             requestIds.add(requestId);
+            if (requestIds.size() > MAX_PARALLEL_REQUESTS) {
+                requestIds.remove(1); // Delete the oldest parallel request id to bound the size of this list.
+            }
             ongoingRequests.put(requestDescriptor, requestIds);
```

Comment:
You're putting this in the cache every time you invoke this method. This is expensive and unnecessary. Consider putting it just after line 173 (first if block).

Also consider using `SimpleCache`.

Reply:
We need to put every new `requestId` in the cache and associate it with the `requestDescriptor`, as the intent is to track every `requestId` that attempts to do some action (in the Controller, so far). For this reason we have 1 `requestDescriptor` that can be associated to multiple concurrent `requestIds`. The only thing I have added is a limit to the amount of `requestIds` that could be associated to a particular `requestDescriptor`, given that in situations in which the Controller cannot complete a given action (e.g., Segment Store is unavailable), the `requestIds` keep accumulating for each retry, thus consuming more heap. I'm not sure if I understand your comment here (specially considering that these are constant time iserts on a map).

I have switched to `SimpleCache` as you have suggested, so we can remove this Guava cache dependency.